src/Controls/src/Core/Handlers/Items/ItemsViewHandler.Windows.cs

```diff
@@ -105,7 +105,52 @@ public static void MapIsVisible(ItemsViewHandler<TItemsView> handler, ItemsView
 
 		public static void MapItemsUpdatingScrollMode(ItemsViewHandler<TItemsView> handler, ItemsView itemsView)
 		{
+			handler.UpdateItemsUpdatingScrollMode();
+		}
+
+		void UpdateItemsUpdatingScrollMode()
+		{
+			if (PlatformView is null || PlatformView.Items is null)
+				return;
+
+			if (VirtualView.ItemsUpdatingScrollMode == ItemsUpdatingScrollMode.KeepScrollOffset)
+			{
+				// The scroll position is maintained when new items are added as the default,
+				// so we don't need to watch for data changes
+				PlatformView.Items.VectorChanged -= OnItemsVectorChanged;
+			}
+			else
+			{
+				PlatformView.Items.VectorChanged += OnItemsVectorChanged;
```

Comment:
Don't we have to unsubscribe and subscribe again, or else we have multiple `OnItemsVectorChanged` registered. The event will fire multiple times for the same change because we subscribed twice or `UpdateItemsUpdatingScrollMode` for some reason run twice.

Reply:
Good point!. Applied changes.